DEV-GUIDE.md

```diff
@@ -209,6 +265,27 @@ Push the containers to your registry:
 make push
 ```
 
+You can selectively build individual Istio containers by using their corresponding
+build targets. Except for mixs, mixc, istioctl, you can prefix *docker.* to the regular build
+targets as described in [Building individual Istio components](#building-individual-istio-components). To build
+mixer server container, use the target *docker.mixer* or *docker.mixer_debug*.
+
+### Building containers with debugger information
```

Comment:
Same as above - everyone will be confused on what containers they use.

We have debug containers and release containers. When configuring you can select a different
one, but we don't want this level of complexity in the build.

Reply:
This is existing already in the Makefile as well, and I'm documenting it in here.
As a developer, I found it's very useful in debugging pilot. I simply build the pilot container, push it up, and restart the pilot pod that will pull the newly-built container. I don't have to build anything else. For example, I use the following to build and debug the pilot container:

```
sudo make docker.pilot DEBUG=all
sudo docker tag pilot:latest baodongli/pilot:debug
sudo docker push baodongli/pilot:debug
```

Reply:
As part of a different change I've been toying with the idea of adding the following to have an eventual replacement for the various scattered push scripts that are invoked by "make push".  The main motivation is to have a push target, e.g., "make HUB=foo TAG=whatever docker.push", though you'd also be able to do a "make HUB=baodongli TAG=debug push.docker.pilot" .  A tradeoff is that the push scripts support comma-separated lists of hubs and tags and I'm hesitant to add the complexity to the Makefile support that.

$(foreach TGT,$(DOCKER_TARGETS),$(eval push.$(TGT): | $(TGT) checkvars; \
        time (docker tag $(subst docker.,,$(TGT)) $(HUB)/$(subst docker.,,$(TGT)):$(TAG) && \
                    docker push $(HUB)/$(subst docker.,,$(TGT)):$(TAG))))

# create a DOCKER_PUSH_TARGETS that's each of DOCKER_TARGETS with a push. prefix
DOCKER_PUSH_TARGETS:=
$(foreach TGT,$(DOCKER_TARGETS),$(eval DOCKER_PUSH_TARGETS+=push.$(TGT)))

# This target pushes each docker image to specified HUB and TAG.
# The push scripts support a comma-separated list of HUB(s) and TAG(s),
# but I'm not sure this is worth the added complexity to support.
docker.push: $(DOCKER_PUSH_TARGETS)

Reply:
Matt: please feel free to delete the multi tag / multi repo complexity.

We do need user to specify one repo and one tag, of course. We don't need a separate flag for each component.

Reply:
With the Makefile as is, plus the DEBUG flag, you can still build individual container with the debugger information. But I will not document it in here since it's not to be advertised yet.